appservice/src/deploy/deployZip.ts

```diff
@@ -101,3 +107,52 @@ async function delayFirstWebAppDeploy(client: SiteClient, asp: AppServicePlan |
         }
     });
 }
+
+async function validateLinuxFunctionAppSettings(client: SiteClient, doBuild: boolean, isConsumption: boolean): Promise<void> {
+    const appSettings: StringDictionary = await client.listApplicationSettings();
+    // tslint:disable-next-line:strict-boolean-expressions
+    appSettings.properties = appSettings.properties || {};
+
+    let hasChanged: boolean = false;
+
+    const keysToRemove: string[] = [
+        'WEBSITE_CONTENTAZUREFILECONNECTIONSTRING',
+        'WEBSITE_CONTENTSHARE'
+    ];
+
+    if (doBuild) {
+        keysToRemove.push(
+            'WEBSITE_RUN_FROM_ZIP',
+            'WEBSITE_RUN_FROM_PACKAGE'
+        );
+    }
+
+    if (!isConsumption) {
+        const dedicatedBuildSettings: [string, string][] = [
+            ['ENABLE_ORYX_BUILD', 'true'],
+            ['SCM_DO_BUILD_DURING_DEPLOYMENT', '1'],
+            ['BUILD_FLAGS', 'UseExpressBuild'],
+            ['XDG_CACHE_HOME', '/tmp/.cache']
+        ];
+
+        for (const [key, value] of dedicatedBuildSettings) {
+            if (!doBuild) {
```

Comment:
This never changes right?  Couldn't you just do a concat and then break out of the for loop?

Reply:
Yeah that would be the exact same - I don't think either of us really care between the two options so Imma leave it